squarelet/organizations/wix.py

```diff
@@ -110,3 +120,100 @@ def sync_wix(organization, plan, user):
+
+    response = requests.post(
+        "https://www.wixapis.com/contacts/v4/contacts",
+        headers=headers,
+        json={
+            "info": {
+                "name": {
+                    "first": first_name,
+                    "last": last_name,
+                },
+                "emails": {
+                    "items": [
+                        {
+                            "email": user.email,
+                            "tag": "MAIN",
+                        }
+                    ]
+                },
+                "company": organization.name,
+            }
+        },
+    )
+    response.raise_for_status()
+    logger.warning(
+        "[WIX-CONTACT] create contact response %d %s",
+        response.status_code,
+        response.json(),
+    )
+    return response.json()["contact"]["id"]
+
+
+def get_contact_by_email_v4(headers, email):
+    """Query for a contact by email using Contacts API v4"""
+    logger.warning("[WIX-CONTACT] get contact by email")
+    response = requests.post(
+        "https://www.wixapis.com/contacts/v4/contacts/query",
+        headers=headers,
+        json={
+            "query": {
+                "filter": {
+                    "info.emails.email": email,
+                }
+            }
+        },
+    )
+    response.raise_for_status()
+    rjson = response.json()
+    logger.warning(
+        "[WIX-CONTACT] get contact by email response %d %s",
+        response.status_code,
+        rjson,
+    )
+    if rjson.get("contacts") and len(rjson["contacts"]) > 0:
+        return rjson["contacts"][0]["id"]
+    else:
+        return None
+
+
+def add_to_waitlist(organization, plan, user):
+    """Add user to waitlist in Wix
+
+    Get or create a contact and apply two labels:
+    - "custom.waitlist" (generic waitlist label)
+    - "custom.waitlist-{plan-slug}" (plan-specific waitlist label)
+    """
+    headers = {
+        "Authorization": settings.WIX_APP_SECRET,
+        "wix-site-id": settings.WIX_SITE_ID,
+    }
+
+    logger.warning(
+        "[WIX-WAITLIST] Adding to waitlist: %s for plan %s", user.email, plan.name
+    )
+
+    try:
+        # Get or create contact using Contacts API
+        contact_id = get_contact_by_email_v4(headers, user.email)
+        if contact_id is None:
+            contact_id = create_contact(headers, organization, user)
+
+        # Add waitlist labels: generic + plan-specific
+        plan_label = f"custom.waitlist-{plan.slug}"
+        response = requests.post(
+            f"https://www.wixapis.com/contacts/v4/contacts/{contact_id}/labels",
+            headers=headers,
+            json={"labelKeys": ["custom.waitlist", plan_label]},
+        )
+        response.raise_for_status()
+        logger.warning("[WIX-WAITLIST] Successfully added to waitlist: %s", user.email)
+    except requests.exceptions.RequestException as exc:
+        logger.error("[WIX-WAITLIST] Failed to add %s to waitlist: %s", user.email, exc)
```

Comment:
When logging an exception, adding `exc_info` as here:
https://github.com/MuckRock/squarelet/blob/master/squarelet/organizations/views.py#L591-L594

Will give Sentry access to the stack trace.